Miscellaneous addons

moonflowerrrr opened the pull request:
Added new subsystem for LEDs
-Added 6 different color options

Added stuff to the dashboard
-Current heading in degrees
-If in field or robot
-Position of module headings in degrees
-Speed of the swerve module
-Left motor and right motor RPM

Added PID controls for the shooter

moonflowerrrr committed:
Wrote a small subsystem for LEDs, adding color presets
-blue bounce
-blue heartbeat
-gold solid
-blue solid

moonflowerrrr committed:
Added useful data to the dashboard
-Module position in degrees
-Module speed
-If field or robot centric
-Robot heading in degrees
-RPM of left and right motor for shooter

moonflowerrrr committed:
Added new subsystem for LEDs
-Added 6 different color options

Added stuff to the dashboard
-Current heading in degrees
-If in field or robot
-Position of module headings in degrees
-Speed of the swerve module
-Left motor and right motor RPM

Added PID controls for the shooter

dlozon committed:
Merge branch 'main' of github.com:TorqueNados-FRC5090/PBOT-2024 into miscellaneous-addons

dlozon commented on src/main/java/frc/robot/subsystems/Blinkin.java:
In the future, we may want to replace these functions with an enumerated list of color profiles and a single function to switch between them

dlozon committed:
removed and renamed several class members from Shooter.java
- removed getMotor functions, we're moving away from providing direct accessors to robot components
- renamed right and left motors to leader and follower
- removed any public references to follower motor
- adjusted the wording slightly on some comments
